desc/coils.py

```diff
@@ -1172,17 +1173,17 @@ def from_symmetry(cls, coils, NFP=1, sym=False):
         # check toroidal extent of coils to be repeated
         maxphi = 2 * np.pi / NFP / (sym + 1)
         data = coils.compute("phi")
+        check_for_intersection = False
         for i, cdata in enumerate(data):
-            errorif(
-                np.any(cdata["phi"] > maxphi),
-                ValueError,
-                f"coil {i} exceeds the toroidal extent for NFP={NFP} and sym={sym}",
-            )
-            warnif(
-                sym and np.any(cdata["phi"] < np.finfo(cdata["phi"].dtype).eps),
-                UserWarning,
-                f"coil {i} is on the symmetry plane phi=0",
+            # check if any coils lie on symmetry planes, these may cause
+            # self-intersection after reflection/rotation
+            maybe_bad_bool = np.any(cdata["phi"] > maxphi) or (
+                sym and np.any(cdata["phi"] < np.finfo(cdata["phi"].dtype).eps)
             )
+            if maybe_bad_bool:
+                # check the coilset for self-intersection before returning coilset
+                # if any lie on symmetry planes
+                check_for_intersection = True
```

Comment:
You are using one bool to set another bool. This could be simplified as: 
```
check_for_intersection = np.any(cdata["phi"] > maxphi) or (
    sym and np.any(cdata["phi"] < np.finfo(cdata["phi"].dtype).eps)
)
```

Reply:
`maybe_bad_bool` is defined at each iteration inside of a for loop, so I need a second bool to check if any of the intermediate bools tripped as True

Reply:
I refactored so that I only use one bool now, it should look cleaner